app/src/main/java/org/p2p/wallet/auth/interactor/restore/UserRestoreInteractor.kt

```diff
@@ -20,17 +20,50 @@ class UserRestoreInteractor(
     }
 
     sealed interface RestoreUserWay {
+        object DevicePlusSocialShareWay : RestoreUserWay
         object SocialPlusCustomShareWay : RestoreUserWay
     }
 
     fun isUserReadyToBeRestored(): Boolean {
-        val isTwoSharesAvailable =
-            restoreFlowDataLocalRepository.socialShare != null && restoreFlowDataLocalRepository.customShare != null
-        return isTwoSharesAvailable
+        val deviceShareCount = countShare(restoreFlowDataLocalRepository.deviceShare) +
+            countShare(restoreFlowDataLocalRepository.socialShare) +
+            countShare(restoreFlowDataLocalRepository.customShare)
+        return deviceShareCount > 1
+    }
+
+    fun isUserReadyToBeRestoredByPhone(): Boolean {
```

Comment:
I think we need connect this methods, and remove useless, let's see way to check isUserReadyForRestore(restoreWay: Object which contains desc of restore way)